docs/docs/reference/contextual/extension-methods.md

```diff
@@ -20,272 +21,245 @@ circle.circumference
+assert(circle.circumference == extension_circumference(circle))
+```
+### Operators
+
+The extension method syntax can also be used to define operators. Examples:
+```scala
+extension (x: String)
+  def < (y: String): Boolean = ...
+extension (x: Elem)
+  def +: (xs: Seq[Elem]): Seq[Elem] = ...
+extension (x: Number)
+  @infix def min (y: Number): Number = ...
+
+"ab" < "c"
+1 +: List(2, 3)
+x min 3
+```
+
+The three definitions above translate to
+```scala
+def extension_< (x: String)(y: String): Boolean = ...
+def extension_+: (xs: Seq[Elem])(x: Elem): Seq[Elem] = ...
+@infix def extension_min(x: Number)(y: Number): Number = ...
+```
+Note the swap of the two parameters `x` and `xs` when translating
+the right-associative operator `+:` to an extension method. This is analogous
+to the implementation of right binding operators as normal methods. The Scala
+compiler preprocesses an infix operation `x +: xs` to `xs.+:(x)`, so the extension
+method ends up being applied to the sequence as first argument (in other words,
+the two swaps cancel each other out).
+
+### Generic Extensions
+
+ It is also possible to extend generic types by adding type parameters to an extension. For instance:
+
+ ```scala
+  extension [T](xs: List[T])
+    def second = xs.tail.head
+
+  extension [T: Numeric](x: T)
+    def + (y: T): T = summon[Numeric[T]].plus(x, y)
+```
+
+If an extension method has type parameters, they come immediately after `extension` and are followed by the extended parameter.
+When calling a generic extension method, any explicitly given type arguments follow the method name. So the `second` method could be instantiated as follows.
+```scala
+  List(1, 2, 3).second[Int]
+```
+Of course, the type argument here would usually be left out since it can be inferred.
+
+
+Extensions can also take using clauses. For instance, the `+` extension above could equivalently be written with a using clause:
+```scala
+  extension [T](x: T)(using n: Numeric[T])
+    def - (y: T): T = n.minus(x, y)
+```
+
+**Note**: Type parameters have to be given after the `extension` keyword;
+they cannot be given after the `def`. This restriction might be lifted in the future once we support multiple type parameter clauses in a method.
+
+### Collective Extensions
+
+Sometimes, one wants to define several extension methods that share the same
+left-hand parameter type. In this case one can "pull out" the common parameters into
+a single extension and enclose all methods in braces or an indented region following a '`:`'.
+Example:
+```scala
+extension (ss: Seq[String]):
+
+  def longestStrings: Seq[String] =
+    val maxLength = ss.map(_.length).max
+    ss.filter(_.length == maxLength)
+
+  def longestString: String = longestStrings.head
+```
+
+Note the right-hand side of `longestString`: it calls `longestStrings` directly, implicitly
+assuming the common extended value `ss` as receiver.
+
+Collective extensions like these are a shorthand for individual extensions
```

Comment:
I would turn this around and say that collective extension methods are the normal syntax and the single definition on one line is the shorthand.

Reply:
The normalization goes from collective to individual.

Reply:
Ok